packages/protocol/test/backward/version.ts

```diff
@@ -0,0 +1,158 @@
+      assert.isFalse(ContractVersion.isValid('test'))
+      assert.isFalse(ContractVersion.isValid('1.a.3.4'))
+    })
+    it('isValid true on valid string', () => {
+      assert.isTrue(ContractVersion.isValid('1.2.3.4'))
+      assert.isTrue(ContractVersion.isValid('-1.0.9.787'))
+      assert.isTrue(ContractVersion.isValid('10.0.224.9'))
+    })
+  })
+  describe('with a new ContractVersion instance', () => {
+    it('properties are well set', () => {
+      const version = new ContractVersion(1, 2, 3, 4)
+      assert.equal(version.storage, 1)
+      assert.equal(version.major, 2)
+      assert.equal(version.minor, 3)
+      assert.equal(version.patch, 4)
+    })
+    it('toString works', () => {
+      const version = new ContractVersion(2, 7, 9, 1)
+      assert.equal(version.toString(), '2.7.9.1')
+    })
+  })
+
+  describe('ContractVersionDelta.fromChanges', () => {
+    it('sets properly the storage Delta', () => {
+      const cvd = ContractVersionDelta.fromChanges(true, false, true, true)
+      assert.equal(cvd.storage, Delta.Increment)
+      assert.equal(cvd.major, Delta.Reset)
+      assert.equal(cvd.minor, Delta.Reset)
+      assert.equal(cvd.patch, Delta.Reset)
+
+      const cvd2 = ContractVersionDelta.fromChanges(false, false, true, true)
+      assert.equal(cvd2.storage, Delta.None)
+      assert.notEqual(cvd2.major, Delta.Reset)
+    })
+
+    it('sets properly the major Delta', () => {
+      const cvd = ContractVersionDelta.fromChanges(false, true, true, true)
+      assert.equal(cvd.major, Delta.Increment)
+      assert.equal(cvd.minor, Delta.Reset)
+      assert.equal(cvd.patch, Delta.Reset)
+
+      const cvd2 = ContractVersionDelta.fromChanges(false, false, true, true)
+      assert.equal(cvd2.major, Delta.None)
+      assert.notEqual(cvd2.minor, Delta.Reset)
+    })
+
+    it('sets properly the minor Delta', () => {
+      const cvd = ContractVersionDelta.fromChanges(false, false, true, true)
+      assert.equal(cvd.minor, Delta.Increment)
+      assert.equal(cvd.patch, Delta.Reset)
+
+      const cvd2 = ContractVersionDelta.fromChanges(false, false, false, true)
+      assert.equal(cvd2.minor, Delta.None)
+      assert.notEqual(cvd2.patch, Delta.Reset)
+    })
+
+    it('sets properly the patch Delta', () => {
+      const cvd = ContractVersionDelta.fromChanges(false, false, false, true)
+      assert.equal(cvd.patch, Delta.Increment)
+
+      const cvd2 = ContractVersionDelta.fromChanges(false, false, false, false)
+      assert.equal(cvd2.patch, Delta.None)
+    })
+  })
+
+  describe('DeltaUtil', () => {
+    it('applyToNumber() works properly', () => {
+      assert.equal(DeltaUtil.applyToNumber(Delta.Reset, 7), 0)
+      assert.equal(DeltaUtil.applyToNumber(Delta.Increment, 9), 10)
+      assert.equal(DeltaUtil.applyToNumber(Delta.None, 11), 11)
+    })
+  })
+
+  describe('with a new ContractVersionDelta instance', () => {
+    it('properties are well set', () => {
```

Comment:
Wording (here and in other cases): try to make these read as a sentence, like "it *does something*". So here maybe something like `it('sets version numbers correctly')`.